React/src/App.jsx

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useState } from 'react';
+import Scheduler from 'devextreme-react/scheduler';
+import { Popup, Position, ToolbarItem } from 'devextreme-react/popup';
+import { isOverlapRecurrentAppointment } from './utils/isOverlapRecurrentAppointment';
+import { defaultData } from './data.js';
+import 'devextreme/dist/css/dx.common.css';
+import 'devextreme/dist/css/dx.material.blue.light.css';
+import './App.css';
+
+const App = () => {
+    const [popupVisible, setPopupVisible] = useState(false);
+    const [data] = useState(defaultData);
+
+    const hideInfo = () => {
+        setPopupVisible(false);
+    }
+
+    const closeButtonOptions = {
+        text: 'Close',
+        onClick: hideInfo,
+    };
+
+    const handleAppointmentActions = (
+        event,
+        recurrentAppointments,
+        newAppointment) => {
+        for (const recurrentAppointment of recurrentAppointments) {
+            const isOverlap = isOverlapRecurrentAppointment(
+                event,
+                recurrentAppointment,
+                newAppointment);
+            if (isOverlap) {
+                cancelAppointmentAdding(event);
+            }
+        }
+    }
+
+    const getRecurrentAppointments = useCallback(() => {
+        return data
+            .filter((appointment) => appointment?.recurrenceRule)
+            .map((appointment) => ({
+                ...appointment,
+                startDate: new Date(appointment.startDate),
+                endDate: new Date(appointment.endDate),
+            }));
+    }, [data]);
+
+    const handleAppointmentAdd = useCallback((event) => {
+        handleAppointmentActions(
+            event,
+            getRecurrentAppointments(),
+            event.appointmentData,
+        );
+    }, [handleAppointmentActions]);
```

Comment:
Let's add ```getRecurrentAppointments``` method to ```useCallback``` dependencies:

```suggestion
    }, [handleAppointmentActions, getRecurrentAppointments]);
```